### PR TITLE
API-91: Add Class

### PR DESCRIPTION
**Pull request by ducvp0702**
Testing result:
image

Before Add Class:
image

After Add Class:
image

### CODE REVIEW

**Review by phantuananh18 — APPROVED** [review]

**Review by Huyle187 — APPROVED** [review]

File: src/Services/GTT/shared/GTT.Infrastructure/Repositories/ClassRepository.cs [hunk 1]
```diff
@@ -69,5 +72,54 @@ public async Task<int> UpdateAsync(Challenge entity)
...
+                parameter.Add("@updatedBy", request.UpdatedBy);
+                parameter.Add("@createdDate", request.CreatedDate);
+                parameter.Add("@updatedDate", request.UpdatedDate);
+                parameter.Add("@IsActive", request.IsActive);
+
+                var result = await _connection.ExecuteAsync(sql, parameter, _tran);
+
+                _tran.Commit();
+
+                return new BaseResponseModel(HttpStatusCode.OK, "Success"); ;
```
Comment (phantuananh18) [inline src/Services/GTT/shared/GTT.Infrastructure/Repositories/ClassRepository.cs]: Remove redundant ";"

**Review by phantuananh18 — APPROVED** [review]

**Review by QuocHuy178 — APPROVED** [review]

File: src/Services/GTT/GTT.Api/ClassManagement/AddClass.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        [OpenApiOperation(nameof(AddClass), "Add Class")]
+        [OpenApiRequestBody(contentType: "application/json", bodyType: typeof(CreateClassRequestModel), Required = true)]
+        [OpenApiResponseWithBody(HttpStatusCode.OK, "application/json", bodyType: typeof(BaseResponseModel))]
+        [OpenApiResponseWithBody(HttpStatusCode.BadRequest, "application/json", bodyType: typeof(BaseResponseModel))]
+        [OpenApiResponseWithoutBody(HttpStatusCode.InternalServerError, Description = "Internal Server Error.")]
+        public async Task<HttpResponseData> AddClassHandler([HttpTrigger(AuthorizationLevel.Function, "post")] HttpRequestData req)
+        {
+            try
+            {
+                _logger.LogInformation("C# HTTP Trigger function CreateExcerciseGroup request.");
```
Comment (QuocHuy178) [inline src/Services/GTT/GTT.Api/ClassManagement/AddClass.cs]: Should change CreateExcerciseGroup  to CreateClass

File: src/Services/GTT/GTT.Api/ClassManagement/AddClass.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                var error = $"[AzureFunction] AddClass - {Helpers.BuildErrorMessage(ex)}";
+                _logger.LogError(error);
+                var response = req.CreateResponse();
+                await response.WriteAsJsonAsync(error, HttpStatusCode.BadRequest);
+                return response;
+            }
+            catch (Exception ex)
+            {
+                var error = $"[AzureFunction] CreateExcerciseGroup - {Helpers.BuildErrorMessage(ex)}";
```
Comment (QuocHuy178) [inline src/Services/GTT/GTT.Api/ClassManagement/AddClass.cs]: Should change CreateExcerciseGroup to CreateClass

File: src/Services/GTT/shared/GTT.Infrastructure/Repositories/ClassRepository.cs [hunk 4]
```diff
@@ -49,25 +53,95 @@ public async Task<int> DeleteAsync(int id)
...
+                parameter.Add("@updatedBy", request.UpdatedBy);
+                parameter.Add("@createdDate", request.CreatedDate);
+                parameter.Add("@updatedDate", request.UpdatedDate);
+                parameter.Add("@IsActive", request.IsActive);
+
+                var result = await _connection.QueryFirstAsync<ClassVM>(sql, parameter, _tran);
+
+                _tran.Commit();
+
+                return new BaseResponseModel(HttpStatusCode.OK, "Success", result);
```
Comment (QuocHuy178) [inline src/Services/GTT/shared/GTT.Infrastructure/Repositories/ClassRepository.cs]: Remove redundant (HttpStatusCode.OK, "Success"), It should be "return new BaseResponseModel(result); is enough

**Pull request merged**